src/command.rs

```diff
@@ -0,0 +1,131 @@
+use std::ffi::OsStr;
+use std::fmt;
+use std::fmt::Debug;
+use std::future::Future;
+use thiserror::Error;
+use tokio::process::Command as TokioCommand;
+
+pub trait HasCommandError {
+    fn title() -> String;
+}
+
+#[derive(Error, Debug)]
+pub enum CommandError<T: fmt::Debug + fmt::Display + HasCommandError> {
+    RunError(std::io::Error),
+    Exit(std::process::Output, String),
+    OtherError(T),
+}
+
+impl<T: fmt::Debug + fmt::Display + HasCommandError> fmt::Display for CommandError<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            CommandError::RunError(err) => {
+                write!(f, "Failed to run {} command: {}", T::title(), err,)
+            }
+            CommandError::Exit(output, cmd) => {
+                let stderr = match String::from_utf8(output.stderr.clone()) {
+                    Ok(stderr) => stderr,
+                    Err(_err) => format!("{:?}", output.stderr),
+                };
+                write!(
+                    f,
+                    "{} command resulted in a bad exit code: {:?}. The failed command is provided below:\n{}\nThe stderr output is provided below:\n{}",
+                    T::title(),
+                    output.status.code(),
+                    cmd,
+                    stderr,
+                )
+            }
+            CommandError::OtherError(err) => write!(f, "{}", err),
+        }
+    }
+}
+
+/// A wrapper over `tokio::process::Command` to provide the `run` method commonly used by `deploy`.
+#[derive(Debug)]
+pub struct Command {
+    pub command: TokioCommand,
+}
+
+impl Command {
+    pub fn new<S: AsRef<OsStr>>(program: S) -> Command {
+        Command {
+            command: TokioCommand::new(program),
+        }
+    }
+
+    pub fn arg<S: AsRef<OsStr>>(&mut self, arg: S) -> &mut Command {
```

Comment:
What is the motivation behind the reimplementation of `arg` and other methods defined for `TokioCommand` rather than creating a single `new` constructor that accepts already constructed `TokioCommand` and `run` to run the command handling the result?